### PR TITLE
[NNC] Add Tests for Dynamic Shape Fusion Change default fusion strategy

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72077
* #72036
* #72038
* #72040
* #72032
* __->__ #71651
* #71650
* #71173

The only tests that regress are because chunk NYI, the other tests that I touched were passing just because the `assertAllFused` wasn't working correctly. That, and we're no longer compiling conv/matmul w dynamic shapes

Differential Revision: [D33801500](https://our.internmc.facebook.com/intern/diff/D33801500)

### CODE REVIEW

File: test/test_jit_fuser_te.py [hunk 1]
```diff
@@ -886,16 +959,26 @@ def test_lstm_traced(self):
...
     def test_milstm(self):
+        if self.dynamic_shapes:
+            self.skipTest("don't run conv with dynamic shapes")
+
         for device in self.devices:
             inputs = get_milstm_inputs(device, training=True)
             module = self.checkScript(MiLSTMCell, inputs)
             forward_graph = module.graph_for(*inputs)
+            # TODO: chunk
+            fusion_group_len = 2 if self.dynamic_shapes else 1
```
Comment (navahgar) [inline test/test_jit_fuser_te.py:979]: Why is the fusion group len for static shapes 1?

**Review by navahgar — APPROVED** [review]
Mostly LGTM.

Title of the PR seems to suggest that the changes are only to the tests. Since you are fixing some issues as well here, can you please update the title to reflect that?

File: test/test_jit_fuser_te.py (hunk 1 above)
Comment (eellison) [reply test/test_jit_fuser_te.py:979]: there is a chunk that isn't being picked up in fusion

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33801500).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33801500).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33801500).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33801500).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33801500).

**Pull request closed without merging**